pkg/discover/cloud_foundry/live_discoverer.go

```diff
@@ -0,0 +1,140 @@
+	}
+	return &LiveDiscovererImpl{cfAPI: kApi.NewCFAPIClient(client, provider.GetKorifiConfig().BaseURL), logger: &log, provider: provider}, nil
+}
+
+func (ld *LiveDiscovererImpl) Discover() (*CloudFoundryManifest, error) {
+	apps, err := ld.cfAPI.ListApps()
+	if err != nil {
+		return nil, fmt.Errorf("error listing CF apps: %v", err)
+	}
+
+	log.Println("Apps discovered:", apps)
+
+	var cfManifest CloudFoundryManifest
+	for _, app := range apps.Resources {
+		log.Println("Processing app:", app.GUID)
+
+		appEnv, err := ld.cfAPI.GetEnv(app.GUID)
+		if err != nil {
+			return nil, fmt.Errorf("error getting environment for app %s: %v", app.GUID, err)
+		}
+
+		appName, err := kApi.GetAppName(*appEnv)
+		if err != nil {
+			return nil, fmt.Errorf("error getting app name: %v", err)
+		}
+
+		normalizedAppName, err := kApi.NormalizeForMetadataName(strings.TrimSpace(appName))
+		if err != nil {
+			return nil, fmt.Errorf("error normalizing app name: %v", err)
+		}
+
+		process, err := ld.cfAPI.GetProcesses(app.GUID)
+		if err != nil {
+			return nil, fmt.Errorf("error getting processes: %v", err)
+		}
+
+		appProcesses := AppManifestProcesses{}
+		for _, proc := range process.Resources {
+			procInstances := uint(proc.Instances)
+
+			appProcesses = append(appProcesses, AppManifestProcess{
+				Type:                         AppProcessType(proc.Type),
+				Command:                      proc.Command,
+				DiskQuota:                    fmt.Sprintf("%d", proc.DiskQuotaMB),
+				HealthCheckType:              AppHealthCheckType(proc.HealthCheck.Type),
+				HealthCheckHTTPEndpoint:      proc.HealthCheck.Data.HTTPEndpoint,
+				HealthCheckInvocationTimeout: uint(proc.HealthCheck.Data.InvocationTimeout),
+				Instances:                    &procInstances,
+				// LogRateLimitPerSecond
+				Memory: fmt.Sprintf("%dMB", proc.MemoryMB),
+				// Timeout
+				// ReadinessHealthCheckType
+				// ReadinessHealthCheckHttpEndpoint
+				// ReadinessHealthInvocationTimeout
+				// ReadinessHealthCheckInterval
+				Lifecycle: string(app.Lifecycle.Type),
+			})
+		}
+
+		routes, err := ld.cfAPI.GetRoutes(app.GUID)
+		if err != nil {
+			return nil, fmt.Errorf("error getting processes: %v", err)
+		}
+		appRoutes := AppManifestRoutes{}
+		for _, r := range routes.Resources {
+			appRoutes = append(appRoutes, AppManifestRoute{
+				Route:    r.URL,
+				Protocol: AppRouteProtocol(r.Protocol),
+				// TODO: Options: loadbalancing?
+			})
+		}
+
+		labels := kApi.ConvertMapToPointer(app.Metadata.Labels)
+		annotations := kApi.ConvertMapToPointer(app.Metadata.Annotations)
+		appManifest := AppManifest{
+			Name: normalizedAppName,
+			Env:  appEnv.EnvironmentVariables,
+			Metadata: &AppMetadata{
+				Labels:      labels,
+				Annotations: annotations,
+			},
+			Processes: &appProcesses,
+			Routes:    &appRoutes,
+			// AppManifestProcess
+			// Buildpacks
+			// RandomRoute
+			// NoRoute
+			// Services
+			// Sidecars
+			// Stack
```

Comment:
@jordigilh Do you have any idea on how to fill the commented ones?

Reply:
I think so. Let's talk tomorrow when you have a moment. I think I've been able to identify the REST endpoints where we can extract all the information from a running app:
https://docs.google.com/document/d/12_LKVU7Fju4-mHafbFxGWtcj2tKBY3MMzL2YY4wp4jI/edit?tab=t.0